lib/primer/view_components/linters/deprecated_components_counter.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require_relative "helpers/rubocop_helpers"
+require_relative "helpers/deprecated_components_helpers"
+
+module ERBLint
+  module Linters
+    # Lints against deprecated components
+    class DeprecatedComponentsCounter < Linter
+      include ERBLint::LinterRegistry
+      include Helpers::RubocopHelpers
+      include Helpers::DeprecatedComponentsHelpers
+
+      def run(processed_source)
+        processed_source.ast.descendants(:erb).each do |erb_node|
+          _, _, code_node = *erb_node
+          code = code_node.children.first.strip
+          node = erb_ast(code)
+
+          next unless node.source.include?("Primer::")
+
+          deprecated_components.each do |component|
+            next unless node.source.include?(component)
+
+            add_offense(
+              erb_node.loc,
+              message(component)
+            )
+          end
+        end
+
+        counter_correct?(processed_source)
+      end
+
+      def autocorrect(processed_source, offense)
+        return unless offense.context
+
+        lambda do |corrector|
+          if processed_source.file_content.include?("erblint:counter #{self.class.name.gsub("ERBLint::Linters::", "")}")
+            # update the counter if exists
+            corrector.replace(offense.source_range, offense.context)
+          else
+            # add comment with counter if none
+            corrector.insert_before(processed_source.source_buffer.source_range, "#{offense.context}\n")
+          end
+        end
+      end
+
+      private
+
+      def counter_correct?(processed_source)
```

Comment:
I saw that `BaseLinter` already defines this. However, `BaseLinter` seems to be tied to a very particular linter implementation, and has a bunch of things that are irrelevant for the check we're trying to do here. Therefore, I am NOT using `BaseLinter` and re-defining the rule. 

I don't understand the `counter_correct?` in `BaseLinter` enough to know whether its refactorable.

Reply:
Update: I removed this definition.

Instead I added `require "erblint-github/linters/custom_helpers"` and pull in the method definition there so we aren't duplicating it.